setuptools/command/egg_info.py

```diff
@@ -239,7 +322,151 @@ def check_broken_egg_info(self):
 
 
 class FileList(_FileList):
-    """File list that accepts only existing, platform-independent paths"""
+    # Implementations of the various MANIFEST.in commands
```

Comment:
Previously, this class was a subclass of the implementation in distutils. It's not clear from this diff if there's still value in subclassing `_FileList`. Is this a complete re-implementation or does it also re-use some of the upstream implementation? Do you expect that these changes could be merged upstream at some point?